zipkin-server/src/main/java/zipkin2/server/internal/throttle/ThrottledStorageComponent.java

```diff
@@ -102,21 +109,47 @@ public ThrottledStorageComponent(StorageComponent delegate, MeterRegistry regist
     return "Throttled{" + delegate.toString() + "}";
   }
 
+  /**
+   * Lazy accesses the request context as we cannot scope the executor used by storage commands per
+   * request.
+   */
+  static final class RequestContextInstrumentedExecutor implements Executor {
```

Comment:
@trustin @kojilin this is what I was asking about in slack.. Basically, I'd like to be able to say `RequestContext.contextAwareExecutor()` (static not instance!)